garak/cli.py

```diff
@@ -64,6 +64,16 @@ def main(arguments=None) -> None:
 
     import argparse
 
+    def worker_count_validation(workers):
+        iworkers = int(workers)
+        if iworkers <= 0:
+            raise argparse.ArgumentTypeError("Need >0 workers (int)" % workers)
+        if iworkers > _config.system.max_workers:
```

Comment:
Testing show this is not really configurable as `_config` has not yet loaded `garak.site.yaml` and also has not loaded `--config` supplied file if passed on cli.

Reply:
I guess, without undue gymnastics, we can either:

1. have a hardcoded, non-configurable cap on CLI param validation (in where? `garak.cli` isn't universal, but otoh `argparse` only applies within this module; top-level values in `garak._config` is the opposite of the direction we're trying to move in)
2. drop CLI `max_workers` validation but let the config-based one be enforced

Having params that are only configurable in `garak.core.yaml` isn't a good option

wdyt? are there other options that make sense? currently leaning toward (2)

Reply:
excellent idea. moved to using non-argparse validation for the cap, and testing for cap config via site yaml.